kernel/kexec.c

```diff
@@ -589,9 +592,12 @@ void kexec_linux(void)
                 if(!bios_mem_map[i].type || bios_mem_map[i].type > 0x1000) {
 			continue;
 		}
-		bios_mem_table[j].addr = bios_mem_map[i].from;
-		bios_mem_table[j].size = bios_mem_map[i].to - bios_mem_map[i].from;
-		bios_mem_table[j].type = bios_mem_map[i].type;
+		bios_mem_table[j].addr = bios_mem_map[i].from_hi;
+		bios_mem_table[j].addr = (bios_mem_table[j].addr << 32) | bios_mem_map[i].from;
+		bios_mem_table[j].size = bios_mem_map[i].to_hi;
+		bios_mem_table[j].size = (bios_mem_table[j].size << 32) | bios_mem_map[i].to;
+		bios_mem_table[j].size -= bios_mem_table[j].addr;
+		bios_mem_table[j].type = bios_mem_map[j].type;
```

Comment:
Ouch! That should be `bios_mem_map[i].type` (i.e. 'i', not 'j').

Reply:
Oh, you're right. Good catch.
Please, submit a PR and I'll merge it.